nodejs/awsx/apigateway/README.md

```diff
@@ -98,6 +98,82 @@ A complete example can be found [here](https://github.com/pulumi/pulumi-awsx/blo
 
 API Gateway can perform basic validations against request parameters, a request payload or both. When a validation fails, a 400 error is returned immediately.
 
+### Lambda Authorizers
+
+[Lambda Authorizers](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-use-lambda-authorizer.html) are AWS Lambda functions that provide control access to an API. You can define a Lambda Authorizer for an Event Handler Route or a Static Route.
+
+To define an Authorizer, you provide a Lambda that fulfills `aws.lambda.EventHandler<AuthorizerEvent, AuthorizerResponse>` or you provide information on a pre-existing Lambda authorizer. The exmaple below shows defining the Authorizer Lambda directly inline. See the [Event Handler Route](#Event-Handler-Route) section on other ways you can define a Lambda for the Authorizer.
+
+```ts
+import * as awsx from "@pulumi/awsx";
+
+const api = new awsx.apigateway.API("myapi", {
+    routes: [{
+        path: "/b",
+        method: "GET",
+        eventHandler: async () => {
+            return {
+                statusCode: 200,
+                body: "<h1>Hello world!</h1>",
+            };
+        },
+        authorizers: [{
+            authorizerName: "prettyAuthorizer",
+            parameterName: "auth",
+            parameterLocation: "query",
+            authType: "custom",
+            authorizer: {
+                type: "request",
+                authorizer: (event, context, callback) => {
+                    console.log("Received event:", JSON.stringify(event, null, 2));
+                    const policy: awsx.apigateway.AuthorizerResponse = {
+                        principalId: "me",
+                        policyDocument: {
+                            Version: "2012-10-17",
+                            Statement: [{
+                                Action: "execute-api:Invoke",
+                                Effect: "Allow",
+                                Resource: event.methodArn,
+                            }],
+                        },
+                    };
+                    callback(null, policy);
```

Comment:
can you see if this will work with the 'async' form?

Reply:
works!